spec/factories/courses.rb

```diff
@@ -60,5 +60,15 @@
         end
       end
     end
+
+    trait :with_editor_by_id do
+      transient do
+        editor_id { nil }
+      end
+
+      after(:build) do |course, evaluator|
+        course.editors << User.find(evaluator.editor_id)
+      end
+    end
```

Comment:
@fosterfarrell9 I vaguely remember that you once offered a better solution to this, but I cannot remember it. Feel free to play around with this and the respective line that starts with `FactoryBot.create("course", "with_editor_by_id",`.

Reply:
I think my solution was for things that appear in the `belongs_to` side of things. If we had the situation that a couse belongs to an editor, then something like `FactoryBot.create(..., editor_id: something)` should work. But since we have `has_many :editors` in `course.rb`, your solution is the correct one in my mind.